docs/OnTrack/Jupyter Notebook/Docker-Documentation-Research-T1-2022.md

```diff
@@ -1,32 +1,67 @@
 # Docker Documentation and Research (WIP) T1-2022
 
-- [Back to Jupyter Notebook Documentation Index](Index.md)
+[Back to Jupyter Notebook Documentation Index](Index.md)
 
-## Conversion Relating to Docker Processes
+## Intro
+
+As we know, the Jupyter Notebook conversion feature will happen via processes that occur inside
+Docker containers. In this document we will discuss the main architecture of this structure in
+relation to OnTrack.
+
+OnTrack is deployed with two main containers that will always be running:
+
+- the frontend container which hosts the logic behind the user interface of OnTrack (mostly
+  unrelated to this feature)
+- the backend container which hosts the logic related to database interactions and other processes
+
+To achieve the Jupyter conversion feature, we will be using a new container that contains all of the
+dependencies (Python, TeX, etc.) needed for a .ipynb to PDF conversion, and performing the
+conversion process within this container. This allows us to have a standalone software package that
+is extendable if required, for example if new Python libraries are required by a unit. It also means
+that we are able to create as many standalone containers as required for different conversion
+processes, such as:
+
+- docx to PDF conversion using Apache POI
+- powerpoint presentation to PDF conversion using Apache POI
+
+This will allow us to add a wide range of compatability in the future, regardless of the software
+requirements.
+
+## Overview of Conversion Process
 
 [<img src="docker_flow.png" />](docker_flow.png)
 
-The API container gets the input file from the frontend. If this input file needs to be converted,
-the API container will first put it into a location **known** to the conversion container (i.e., a
-_temp_ directory in a shared volume). The shared _temp_ directory ensures that both the API
-container **and** the conversion container have read and write access to the input and output files.
-Note that this _temp_ directory will be the shared volume specified in `docker-compose.yml`.
+When the OnTrack frontend sends a new file to the OnTrack backend, the backend will be able to
+determine the file type and if it needs to be converted. If the OnTrack backed recieves a .ipynb
+file, it will 'spin up' the Jupyter to PDF conversion container to perform the conversion process.
```

Comment:
Reader may not understand what you mean by "spin up". Explanation of this may improve readability. Just a suggestion though.